src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CaseReferenceValidator.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd;
+
+import io.vavr.control.Option;
```

Comment:
Hmm. Just now paying attention to `vavr` usage over here in this repository. As I scroll in this file I see common usage of `java.util.Optional`. Is there a benefit of using extra library just to have some sort of feature going on? I do get the `cats` for `scala` just questioning `flatmap` with friends usage over here

Reply:
if we are using the Optional from java 9 then all would be good the problem is that Optional in java 8 does not have  Optional<T> orElse ( Optional<T> ) 
this means that the solution mentioned in the comments above can be used.
I guess I can back port this if its required.

Reply:
Ive moved it back to Optional since Vavr Option might not go down well here.